imap_processing/codice/codice_l1a_lo_angular.py

```diff
@@ -0,0 +1,359 @@
+    )
+
+    # ========== Get LUT Data ===========
+    # Read information from LUT
+    sci_lut_data = read_sci_lut(lut_file, table_id)
+
+    view_tab_info = get_view_tab_info(sci_lut_data, view_id, apid)
+    view_tab_obj = ViewTabInfo(
+        apid=apid,
+        view_id=view_id,
+        sensor=view_tab_info["sensor"],
+        three_d_collapsed=view_tab_info["3d_collapse"],
+        collapse_table=view_tab_info["collapse_table"],
+    )
+
+    if view_tab_obj.sensor != 0:
+        raise ValueError("Unsupported sensor ID for Lo angular processing.")
+
+    # ========= Decompress and Reshape Data ===========
+    # Lookup SW or NSW species based on APID
+    if view_tab_obj.apid == CODICEAPID.COD_LO_SW_ANGULAR_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["angular"]["sw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-sw-angular"
+    elif view_tab_obj.apid == CODICEAPID.COD_LO_NSW_ANGULAR_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["angular"]["nsw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-nsw-angular"
+    else:
+        raise ValueError(f"Unknown apid {view_tab_obj.apid} in Lo species processing.")
+
+    compression_algorithm = constants.LO_COMPRESSION_ID_LOOKUP[view_tab_obj.view_id]
+    # Decompress data using byte count information from decommed data
+    binary_data_list = unpacked_dataset["data"].values
+    byte_count_list = unpacked_dataset["byte_count"].values
+
+    # The decompressed data in the shape of (epoch, n). Then reshape later.
+    decompressed_data = [
+        decompress(
+            packet_data[:byte_count],
+            compression_algorithm,
+        )
+        for (packet_data, byte_count) in zip(
+            binary_data_list, byte_count_list, strict=False
+        )
+    ]
+
+    # Look up collapse pattern using LUT table. This should return collapsed shape.
+    collapsed_shape = get_collapse_pattern_shape(
+        sci_lut_data, view_tab_obj.sensor, view_tab_obj.collapse_table
+    )
+
+    # Reshape decompressed data to:
+    #   (num_packets, num_species, esa_steps, 12, 5)
+    # 24 includes despinning spin sector. Then at later steps,
+    # we handle despinning.
+    num_packets = len(binary_data_list)
+    esa_steps = constants.NUM_ESA_STEPS
+    num_species = len(species_names)
+    species_data = np.array(decompressed_data).reshape(
+        num_packets, num_species, esa_steps, *collapsed_shape
+    )
+
+    # Despinning
+    # ----------------
+    species_data = _despin_species_data(species_data, sci_lut_data, view_tab_obj)
+
+    # ========== Get Voltage Data from LUT ===========
+    # Use plan id and plan step to get voltage data's table_number in ESA sweep table.
+    # Voltage data is (128,)
+    esa_table_number = (
+        sci_lut_data["plan_tab"].get(f"({plan_id}, {plan_step})").get("lo_stepping")
```

Comment:
Rather than get() throughout could you do the Key access way? I think it will be easier to read KeyErrors than None type does not have attribute .get() and you don't know where it is failing.